runner/src/mill/runner/MillMain.scala

```diff
@@ -91,8 +125,8 @@ object MillMain {
       systemExit: Int => Nothing,
       serverDir: os.Path
   ): (Boolean, RunnerState) = {
-    val streams = streams0
-    mill.define.SystemStreams.withStreams(streams) {
+    val bspMode = args.headOption.contains("--bsp")
```

Comment:
I removed this check in the earlier BSP cleanup because it wasn't clear what the purpose of this check was, and it appears this code path was not being triggered anyway when I imported the project into IntelliJ.

What's the purpose of this check when we already do the stream redirection after argument parsing below?

Reply:
> What's the purpose of this check when we already do the stream redirection after argument parsing below?

If all goes well, that's even. But if anything goes wrong, in particular while debugging / developing on Mill, it's better to have the redirection in place early, to make it less likely to miss anything.

Plus it wasn't done right after argument parsing before this PR, but a bit after that, which increases the odd of messages going to the wrong stream.